app/services/aws_sqs_poller.rb

```diff
@@ -7,30 +7,38 @@ def self.poll
 
     poller.poll do |msg|
       message_body = JSON.parse(msg[:body])
-      next unless message_body["job_instance"]["name"] == Rails.configuration.alma[:full_dump_job_name]
-      dump = AlmaFullDumpFactory.full_bib_dump(message_body)
+      job_name = message_body["job_instance"]["name"]
+      next unless jobs_map.keys.include?(job_name)
+      dump = AlmaDumpFactory.bib_dump(
+        message_body,
+        jobs_map[job_name]
+      )
       # running dump creation in the background prevents the queue
       # event from timing out and requeuing
-      AlmaFullDumpTransferJob.perform_later(
+      AlmaDumpTransferJob.perform_later(
         dump: dump,
         job_id: message_body["job_instance"]["id"]
       )
     end
   end
 end
 
-class AlmaFullDumpFactory
+def jobs_map
+  @jobs_map ||= Rails.configuration.alma[:job_name_record_type]
+end
+
+class AlmaDumpFactory
   attr_reader :message
   def initialize(message)
     @message = message
   end
 
-  def self.full_bib_dump(message)
-    new(message).full_bib_dump
+  def self.bib_dump(message, constant)
+    new(message).bib_dump(constant)
```

Comment:
Question: Does one message create multiple kinds of bib dump constants? Should constant be in the initializer rather than an argument to `bib_dump`? Or even better be able to be derived from the message given, so we can remove passing that argument around?